modules/decklinksrc/decklinksrc.py

```diff
@@ -28,28 +31,40 @@ class DeckLinkSRC:
         Displays live video feed of DeckLink stream
     """
 
-    def __init__(self):
+    def __init__(self, videoPipeline):
         """
         Initializes DeckLink video stream
         """
         self.__currentFrame = None
-        self.capture = cv2.VideoCapture('decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink') #Starts capture on initialization of object
+        self.capture = cv2.VideoCapture(
+            'decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink')  # Starts capture on initialization of object
+        self.videoPipeline = videoPipeline
+        mainProcess = mp.Process(target=self._mainProcess_, daemon=True)
+        mainProcess.start()
+
+    def _mainProcess_(self):
+        print("decklink started")
+        while True:
+            newFrame = self.grab()
+            if self.videoPipeline.empty() or newFrame != self.videoPipeline.get():
+                self.videoPipeline.put(newFrame)
+            sleep(0.1)
```

Comment:
Any reason for the sleep?

Reply:
Basically just so I don't flood the system, I'm not sure how good python is with memory management but I worry filling the pipeline too fast will cause issues